warehouse/manage/views.py

```diff
@@ -1443,66 +1455,149 @@ def manage_project_roles(project, request, _form_class=CreateRoleForm):
                 f"address and cannot be added as a {role_name} for project",
                 queue="error",
             )
-        else:
-            request.db.add(
-                Role(user=user, project=project, role_name=form.role_name.data)
+        elif (
+            user_invite
+            and user_invite.invite_status == RoleInvitationStatus.Pending.value
+        ):
+            request.session.flash(
+                f"User '{username}' already has an active invite. "
+                f"Please try again later.",
+                queue="error",
             )
+        elif (
+            user_invite
+            and user_invite.invite_status == RoleInvitationStatus.Accepted.value
+        ):
+            request.session.flash(
+                f"User '{username}' is already a collaborator on the project.",
+                queue="error",
+            )
```

Comment:
What happens here if we're just changing the user's role on the project (e.g. from Maintainer to Owner), so there's already one accepted invitation, but we need to generate another with a different role?